python/http_ece/__init__.py

```diff
@@ -86,47 +148,141 @@ def lengthPrefix(key):
         info=nonceinfo,
         backend=default_backend()
     )
-    result = (hkdf_key.derive(secret), hkdf_nonce.derive(secret))
-    return result
+    return hkdf_key.derive(secret), hkdf_nonce.derive(secret)
+
 
 def iv(base, counter):
+    """Generate an initialization vector.
+
+    """
     if (counter >> 64) != 0:
-        raise Exception(u"Counter too big")
+        raise ECEException(u"Counter too big")
     (mask,) = struct.unpack("!Q", base[4:])
     return base[:4] + struct.pack("!Q", counter ^ mask)
 
-def decrypt(buffer, salt, key=None, keyid=None, dh=None, rs=4096, authSecret=None, padSize=2):
-    def decryptRecord(key, nonce, counter, buffer):
+
+def decrypt(content, salt, key=None, keyid=None, dh=None, rs=4096,
+            auth_secret=None, version="aesgcm", **kwargs):
+    """
+    Decrypt a data block
+
+    :param content: Data to be decrypted
+    :type content: str
+    :param salt: Encryption salt
+    :type salt: str
+    :param key: local public key
+    :type key: str
+    :param keyid: Internal key identifier for private key info
+    :type keyid: str
+    :param dh: Remote Diffie Hellman sequence
+    :type dh: str
```

Comment:
In the JS code I read the DH key from the keyid in the header when I'm using DH.